### PR TITLE
Add recognition from VertexAI

### PR DESCRIPTION
**Pull request by jcorrado76**
added code to deploy fine-tuned Resnet 50 to Vertex AI, and make predictions from it

### CODE REVIEW

File: 3_computer_vision/ml_project_4_face_detection/README.md [hunk 1]
```diff
@@ -90,11 +90,107 @@ make app
...
+Then, unzip that archive:
+
+```bash
+unzip celebrity-18.zip
+```
+
+This will create a folder called `Celebrity Faces Dataset` on your machine.
+
+
+We'll be using AWS Rekognition for celebrity facial recognition, as they provide this out of the box, and the SDK is very easy to use.
```
Comment (abhishek-kumar) [inline 3_computer_vision/ml_project_4_face_detection/README.md]: Please mention here that this is just part 1, we will later (below) train a new model.

File: 3_computer_vision/ml_project_4_face_detection/backend/recognition.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+) -> Optional[str]:
+    logger.info(f"Input image: {image}")
+    example_image_array: np.ndarray = img_to_array(image)
+    logger.info(f"Image array shape: {example_image_array.shape}")
+    example_image_array = np.expand_dims(example_image_array, axis=0)
+    logger.info(f"After expanding dimensions: {example_image_array.shape}")
+    example_image_array = preprocess_input(example_image_array)
+    logger.info(f"After preprocessing: {example_image_array.shape}")
+
+    prediction_probabilities = model.predict(example_image_array)
```
Comment (abhishek-kumar) [inline 3_computer_vision/ml_project_4_face_detection/backend/recognition.py:87]: Something to consider for fun: you could also output the second and third predictions in addition to the argmax, and show that these can be used for further training (as negative examples). This helps improve convergence.

File: 3_computer_vision/ml_project_4_face_detection/backend/recognition.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    Taken from https://cloud.google.com/vertex-ai/docs/predictions/get-online-predictions#predict-request
+
+    Live endpoints are used like this:
+    endpoint.predict(instances=[[6.7, 3.1, 4.7, 1.5], [4.6, 3.1, 1.5, 0.2]])
+    """
+    aiplatform.init(project=project, location=location)
+
+    # endpoint = aiplatform.Endpoint(endpoint_name=endpoint_id)
+    endpoint = aiplatform.Endpoint(
+        endpoint_name="projects/820240006900/locations/us-east4/endpoints/5089357849998393344"
```
Comment (abhishek-kumar) [inline 3_computer_vision/ml_project_4_face_detection/backend/recognition.py]: Could we make this a constant or a flag at the top with instructions on how to update it?

File: 3_computer_vision/ml_project_4_face_detection/backend/test_recognition.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+@pytest.fixture
+def test_pil_image() -> PIL.Image:
+    example_file_path = pathlib.Path(".test_assets/test_image.jpg")
+    loaded_img_from_file: PIL.Image = load_image_from_file(
+        image_path=example_file_path,
+    )
+    return loaded_img_from_file
+
+
+@pytest.fixture
```
Comment (abhishek-kumar) [inline 3_computer_vision/ml_project_4_face_detection/backend/test_recognition.py:40]: Nice tests!

**Review by abhishek-kumar — APPROVED** [review]
Thank you! Could you make a post on slack as well?

File: 3_computer_vision/ml_project_4_face_detection/backend/recognition.py (hunk 3 above)
Comment (jcorrado76) [reply 3_computer_vision/ml_project_4_face_detection/backend/recognition.py]: refactored it as a global constant, and added instructions to README

**Pull request merged**